src/arch/ia32/interrupts/pic8259a.rs

```diff
@@ -0,0 +1,248 @@
+    MASTER,
+}
+
+/// Struct representing a 8259 PIC
+pub struct PIC8259a {
+    /// Command port of the chip.
+    com_port: Port<u8>,
+    /// Data port of the chip.
+    data_port: Port<u8>,
+    /// Pin connected to Master/Slave. None if single mode.
+    pin: Option<u8>,
+    /// PIC IRQs offset in Interrupt Descriptor Table.
+    offset: u8,
+    /// PIC Role.
+    role: PicRole,
+}
+
+impl PIC8259a {
+    /// Creates a [`PIC8259a`]'s representation in Single mode.
+    ///
+    /// # Arguments
+    ///
+    /// * `addr` - PIC Command port address. Data port address will be deduced.
+    /// * `offset` - PIC IRQs offset in Interrupt Descriptor Table.
+    const fn single(addr: u16, offset: u8) -> Self {
+        if offset % 8 != 0 {
+            panic!("8259a: offset must be a multiple of 8.");
+        }
+
+        Self {
+            com_port: Port::new(addr),
+            data_port: Port::new(addr + 1),
+            pin: None,
+            offset: offset,
+            role: PicRole::SINGLE,
+        }
+    }
+
+    /// Creates a [`PIC8259a`]'s representation in Master mode.
+    ///
+    /// # Arguments
+    ///
+    /// * `addr` - PIC Command port address. Data port address will be deduced.
+    /// * `slave_pin` - The pin on which the PIC slave is connected to.
+    /// * `offset` - PIC IRQs offset in Interrupt Descriptor Table.
+    pub const fn master(addr: u16, slave_pin: u8, offset: u8) -> Self {
+        if slave_pin >= 8 {
+            panic!("8259a: pin must be between 0 and 7");
+        }
+
+        if offset % 8 != 0 {
+            panic!("8259a: offset must be a multiple of 8.");
+        }
+
+        Self {
+            com_port: Port::new(addr),
+            data_port: Port::new(addr + 1),
+            pin: Some(slave_pin),
+            offset: offset,
+            role: PicRole::MASTER,
+        }
+    }
+
+    /// Creates a [`PIC8259a`]'s representation in Slave mode.
+    ///
+    /// # Arguments
+    ///
+    /// * `addr` - PIC Command port address. Data port address will be deduced.
+    /// * `slave_pin` - The pin on which the master PIC is connected to.
+    /// * `offset` - PIC IRQs offset in Interrupt Descriptor Table.
+    pub const fn slave(addr: u16, slave_pin: u8, offset: u8) -> Self {
+        if slave_pin >= 8 {
+            panic!("8259a: pin must be between 0 and 7");
+        }
+
+        if offset % 8 != 0 {
+            panic!("8259a: offset must be a multiple of 8.");
+        }
+
+        Self {
+            com_port: Port::new(addr),
+            data_port: Port::new(addr + 1),
+            pin: Some(slave_pin),
+            offset: offset,
+            role: PicRole::SLAVE,
+        }
+    }
+
+    /// Sends the Initialisation Command Word 1.
+    ///
+    /// # Arguments
+    ///
+    /// * `has_icw4` - If true, ICW4 will be expected.
+    /// * `adi` - CALL Address Inverval (true = 4, false = 8).
+    /// * `ltim` - Level Triggered mode (true) or Edge Triggered mode (false).
```

Comment:
I think we want to make this variable's name more explicit. Maybe even use an enum with two values

```rust
enum TriggerMode {
    LEVEL,
    EDGE,
};

```

Reply:
Fixed in 1b7fa082f519efe83b62b109f000ef7bec0b1cda